### DISCUSSION TITLE
How to make component resizable?

### DISCUSSION THREAD

**Discussion by saudAtIrisdame** [GrapesJS/grapesjs#4014] [Q&A]
I want to resize component just like the image component

**Comment by aliibrahim123** [comment]
you can set `resizable` property to `true`

**Reply by saudAtIrisdame** [reply]
@aliibrahim123 
i am using built-in components. how can I make built-in components resizable?

**Reply by aliibrahim123** [reply]
you can update it like this
```
editor.DomComponents.addType('text', {
    model: editor.DomComponents.getType('text').model.extend({
      defaults: Object.assign({}, editor.DomComponents.getType('text').model.prototype.defaults, {
        droppable: true,
        resizable:true
      },{
        isComponent : false,
      } )
    }),
    view: editor.DomComponents.getType('text').view
  });
```
this example is text
you can change it to anything

**Reply by saudAtIrisdame** [reply]
thanks @aliibrahim123

**Reply by DykiSA** [reply]
Hi @aliibrahim123, I did put your snippets into this [pen](https://codepen.io/DykiSA/pen/ExpvPJB). But it is not working due to this error:
```
Uncaught TypeError: Cannot set property defaults of #<n> which has only a getter
    at grapesjs:2:926798
```

Seems defaults can't be reset.

**Reply by crazyursus92** [reply]
I found this solution.

```
editor.on("component:selected", (component) => {
    component.set("resizable", true);
  });
```

It works with grapesjs v0.21.1 for sure. And you can easily add some condition to add resize for some specific component type

**Reply by artf** [reply]
This is all you need if you have to extend a component definition
```js
editor.Components.addType('text', {
    model: {
      defaults: { resizable: true },
    },
 });
```
